File: amic-platform/src/modules/ma/types/permit.ts
export type PermitFilingType =
  | "CHANGE_NOTIFICATION"
  | "CHANGE_APPROVAL"
  | "NEW_REGISTRATION"
  | "RENEWAL";

export type PermitTimingType = "PRE_FILING" | "POST_FILING" | "BOTH";

export type PermitAnalysisStatus =
  | "PENDING"
  | "ANALYZING"
  | "COMPLETED"
  | "FAILED"
  | "MANUALLY_REVIEWED";

export type PermitRequirementStatus =
  | "IDENTIFIED"
  | "DOCUMENTS_PREPARING"
  | "FILED"
  | "APPROVED"
  | "NOT_APPLICABLE";

export interface ExistingPermit {
  name: string;
  issuer: string;
  reg_number?: string;
}

export interface RequiredDocument {
  name: string;
  description?: string;
}

export interface PermitAnalysis {
  id: string;
  transaction_id: string;
  status: PermitAnalysisStatus;
  business_types: string[] | null;
  existing_permits: ExistingPermit[] | null;
  analysis_method: string | null;
  llm_cost_usd: number | null;
  analysis_notes: string | null;
  analyzed_by_email: string | null;
  created_at: string;
  updated_at: string;
}

export interface PermitRequirement {
  id: string;
  analysis_id: string;
  transaction_id: string;
  permit_name: string;
  regulatory_body: string;
  legal_basis: string | null;
  filing_type: PermitFilingType;
  timing_type: PermitTimingType;
  pre_filing_deadline_days: number | null;
  post_filing_deadline_days: number | null;
  calculated_deadline: string | null;
  required_documents: RequiredDocument[] | null;
  status: PermitRequirementStatus;
  source: string;
  confidence: number | null;
  compliance_item_id: string | null;
  notes: string | null;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface PermitAnalyzeRequest {
  business_types: string[];
  existing_permits: ExistingPermit[];
}

export interface PermitRequirementUpdate {
  status?: PermitRequirementStatus;
  notes?: string;
}

export interface IndustryOption {
  code: string;
  label: string;
  sub_categories: string[] | null;
}

File: deal-mgmt/app/schemas/permit.py
"""인허가 분석 Pydantic 스키마."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import (
    PermitAnalysisStatus,
    PermitFilingType,
    PermitRequirementStatus,
    PermitTimingType,
)

# ── 공통 서브 모델 ──────────────────────────────────────────

class ExistingPermit(BaseModel):
    name: str = Field(..., min_length=1, max_length=300)
    issuer: str = Field(..., min_length=1, max_length=200)
    reg_number: str | None = Field(None, max_length=100)


class RequiredDocument(BaseModel):
    name: str
    description: str | None = None


# ── PermitAnalysis ──────────────────────────────────────────

class PermitAnalysisOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    transaction_id: uuid.UUID
    status: PermitAnalysisStatus
    business_types: list[str] | None = None
    existing_permits: list[ExistingPermit] | None = None
    analysis_method: str | None = None
    llm_cost_usd: float | None = None
    analysis_notes: str | None = None
    analyzed_by_email: str | None = None
    created_at: datetime
    updated_at: datetime


class PermitAnalyzeRequest(BaseModel):
    """인허가 분석 실행 요청."""
    business_types: list[str] = Field(..., min_length=1)
    existing_permits: list[ExistingPermit] = Field(default_factory=list)


# ── PermitRequirement ───────────────────────────────────────

class PermitRequirementOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    analysis_id: uuid.UUID
    transaction_id: uuid.UUID
    permit_name: str
    regulatory_body: str
    legal_basis: str | None = None
    filing_type: PermitFilingType
    timing_type: PermitTimingType
    pre_filing_deadline_days: int | None = None
    post_filing_deadline_days: int | None = None
    calculated_deadline: str | None = None
    required_documents: list[RequiredDocument] | None = None
    status: PermitRequirementStatus
    source: str
    confidence: float | None = None
    compliance_item_id: uuid.UUID | None = None
    notes: str | None = None
    sort_order: int
    created_at: datetime
    updated_at: datetime


class PermitRequirementCreate(BaseModel):
    """인허가 요건 수동 추가."""
    permit_name: str = Field(..., min_length=1, max_length=300)
    regulatory_body: str = Field(..., min_length=1, max_length=200)
    legal_basis: str | None = Field(None, max_length=500)
    filing_type: PermitFilingType
    timing_type: PermitTimingType
    pre_filing_deadline_days: int | None = None
    post_filing_deadline_days: int | None = None
    required_documents: list[RequiredDocument] | None = None
    notes: str | None = None


class PermitRequirementUpdate(BaseModel):
    """인허가 요건 수정."""
    status: PermitRequirementStatus | None = None
    notes: str | None = None
    required_documents: list[RequiredDocument] | None = None


# ── KB 참조 ─────────────────────────────────────────────────

class IndustryOption(BaseModel):
    code: str
    label: str
    sub_categories: list[str] | None = None
